From: Marco Martin <notmart@gmail.com>
Date: Thu, 02 Apr 2015 12:25:56 +0000
Subject: iexplicitly fallback to unknown icon
X-Git-Url: http://quickgit.kde.org/?p=plasma-desktop.git&a=commitdiff&h=9ba1ef4777aae9a2492f9960d5fc46b3a6415d85
---
iexplicitly fallback to unknown icon

as in kicker, we have to manually fallback now
BUG:345740
---


--- a/applets/kickoff/core/applicationmodel.cpp
+++ b/applets/kickoff/core/applicationmodel.cpp
@@ -149,7 +149,7 @@
             }
 
             AppNode *newnode = new AppNode();
-            newnode->icon = QIcon::fromTheme(p->icon());
+            newnode->icon = QIcon::fromTheme(p->icon(), QIcon::fromTheme("unknown"));
             newnode->appName = p->name();
             newnode->genericName = p->genericName();
             newnode->desktopEntry = p->entryPath();
@@ -264,7 +264,7 @@
 
         AppNode *newnode = new AppNode();
         newnode->iconName = icon;
-        newnode->icon = QIcon::fromTheme(icon);
+        newnode->icon = QIcon::fromTheme(icon, QIcon::fromTheme("unknown"));
         newnode->appName = appName;
         newnode->genericName = genericName;
         newnode->relPath = relPath;

--- a/applets/kickoff/core/models.cpp
+++ b/applets/kickoff/core/models.cpp
@@ -77,7 +77,7 @@
         item = new QStandardItem;
         KDesktopFile desktopFile(url.toLocalFile());
         item->setText(QFileInfo(urlString.mid(0, urlString.lastIndexOf('.'))).completeBaseName());
-        item->setIcon(QIcon::fromTheme(desktopFile.readIcon()));
+        item->setIcon(QIcon::fromTheme(desktopFile.readIcon(), QIcon::fromTheme("unknown")));
 
         //FIXME: desktopUrl is a hack around borkage in KRecentDocuments which
         //       stores a path in the URL field!
@@ -107,7 +107,7 @@
 
         item->setText(basename);
         //FIXME
-//         item->setIcon(QIcon::fromTheme(KMimeType::iconNameForUrl(url)));
+//         item->setIcon(QIcon::fromTheme(KMimeType::iconNameForUrl(url), QIcon::fromTheme("unknown")));
         item->setData(url.url(), Kickoff::UrlRole);
         item->setData(subTitle, Kickoff::SubTitleRole);
 
@@ -150,7 +150,7 @@
     QString appName = service->name();
     bool nameFirst = displayOrder == NameBeforeDescription;
     appItem->setText(nameFirst || genericName.isEmpty() ? appName : genericName);
-    appItem->setIcon(QIcon::fromTheme(service->icon()));
+    appItem->setIcon(QIcon::fromTheme(service->icon(), QIcon::fromTheme("unknown")));
     appItem->setData(service->entryPath(), Kickoff::UrlRole);
 
     if (nameFirst) {

